src/basic/fd-util.c

```diff
@@ -928,6 +928,18 @@ int rearrange_stdio(int original_input_fd, int original_output_fd, int original_
         return r;
 }
 
+int fd_verify_executable(int fd) {
+        struct stat st;
+
+        if (fstat(fd, &st) < 0)
+                return -errno;
+
+        if (!(st.st_mode & S_IXUSR))
+                return -EACCES;
```

Comment:
that's quite a different test than access() does.

instead:

```c
access("/proc/self/fd/%i", X_OK);
```

In contrast to the fstat() check this actually checks if *we* can execute the thing.

Reply:
Thanks, I missed that helper.